noir-projects/aztec-nr/aztec/src/utils/with_hash.nr

```diff
@@ -0,0 +1,121 @@
+use crate::{
+    context::{PublicContext, UnconstrainedContext},
+    history::public_storage::PublicStorageHistoricalRead,
+    oracle,
+};
+use dep::protocol_types::{
+    address::AztecAddress, block_header::BlockHeader, hash::poseidon2_hash, traits::Packable,
+};
+
+pub struct WithHash<T, let N: u32> {
```

Comment:
It'd be nice to have an explanation here for what this is

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/8022/commits/fdc1c84e8ea0a433e1e5cc3901b1a292d3e95033